### PR TITLE
[NASA] [Feature] Guarding against unimplemented configuration (#40)

### PR DESCRIPTION
**Pull request by FlorianDeconinck**
Guarding against unimplemented namelists options:
- a2b_ord4
- d_sw
- fv_dynamics
- fv_subgridz
- neg_adj3
- divergence damping
- xppm
- yppm

Misc:
- Fix `netcdf_monitor` not mkdir the directory
- Add `as_dict` to the dycore state to dump the dycore more easily

### CODE REVIEW

File: fv3core/pace/fv3core/stencils/fv_subgridz.py [hunk 1]
```diff
@@ -783,12 +782,17 @@ def __init__(
...
+                "DryConvectiveAdjustment (fv_subgridz):"
+                " Hydrostatic is not implemented"
+            )
         grid_indexing = stencil_factory.grid_indexing
         self._k_sponge = n_sponge
-        if self._k_sponge is not None:
-            if self._k_sponge < 3:
-                return
+        if self._k_sponge is not None and self._k_sponge < 3:
+            raise NotImplementedError(
```
Comment (oelbert) [inline fv3core/pace/fv3core/stencils/fv_subgridz.py:793]: This should be a value error not a not implemented error, the model shouldn't have nsponge < 3

File: fv3core/pace/fv3core/stencils/remap_profile.py [hunk 2]
```diff
@@ -591,7 +591,11 @@ def __init__(
             config=stencil_factory.config.dace_config,
         )
 
-        assert kord <= 10, f"kord {kord} not implemented."
+        if kord > 10:
```
[the remark this answers is not in the document]
Comment (oelbert) [reply fv3core/pace/fv3core/stencils/remap_profile.py:594]: not necessarily here, just in future. Maybe log it?

**Review by oelbert — CHANGES_REQUESTED** [review]
Minor changes, otherwise good.

File: fv3core/pace/fv3core/stencils/remap_profile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (FlorianDeconinck) [reply fv3core/pace/fv3core/stencils/remap_profile.py:594]: https://github.com/NOAA-GFDL/pace/issues/49

File: fv3core/pace/fv3core/stencils/fv_subgridz.py (hunk 1 above)
Comment (FlorianDeconinck) [reply fv3core/pace/fv3core/stencils/fv_subgridz.py]: Done

**Review by oelbert — APPROVED** [review]
👍 thanks for changes

**Review by fmalatino — APPROVED** [review]

**Review by bensonr — APPROVED** [review]

**Pull request merged**